src/registry/addressNL/edit.tsx

```diff
@@ -81,6 +142,15 @@ const EditForm: EditFormDefinition<AddressNLComponentSchema> = () => {
   const [isKeyManuallySetRef, generatedKey] = useDeriveComponentKey();
   const {hasAnyError} = useErrorChecker<AddressNLComponentSchema>();
   Validate.useManageValidatorsTranslations<AddressNLComponentSchema>(['required']);
+  Validate.useManageValidatorsTranslations(
+    ['pattern'],
+    `openForms.components.postcode.translatedErrors`
```

Comment:
this doesn't match with the type passed in for the generic [here](https://github.com/open-formulieren/formio-builder/pull/167/files#diff-e1c3cd11dc4c171ab1f0efa43e03b841daaaab0141ce38725838bd6272ec462dR16) - and you're also not passing the sub-schema generic argument to `useManageValidatorsTranslations`

Should be something like:

```tsx
type AddressSubComponents = Required<AddressNLComponentSchema>['openForms']['components'];

Validate.useManageValidatorsTranslations<AddressSubComponents[keyof AddressSubComponents]>(...);
```

I also notice that the field name always ends with `translatedErrors`, so perhaps it's better to pass in an optional prefix for deeply nested fields instead of the entire field name? Then the generic using `S['translatedErrors']` would still enforce the same/similar shape of managing translation errors, which is nice to have.

Reply:
I'm trying to get this to work, but it's costing me quite a lot of time. I've got it "working" (no typing errors, except the build does fail) like this with these changes in types https://github.com/open-formulieren/types/pull/52/files:

```typescript
Validate.useManageValidatorsTranslations<NonNullable<AddressComponents[keyof AddressComponents]>['validate']>(
    ['pattern'], `openForms.components.postcode.translatedErrors`
  );
```

But then when building, I get:

```
src/registry/addressNL/edit.tsx:340:20 - error TS2322: Type '{ pattern: string; }' is not assignable to type 'HasValidation<"pattern">'.
  Object literal may only specify known properties, and 'pattern' does not exist in type 'HasValidation<"pattern">'.

340         validate: {pattern: ''},
```

Any ideas? :grimacing:

Reply:
not immediately, I'll have to dive into this, I'll reserve some time tomorrow :grimacing: